### PR TITLE
[tune] Remove py2.7-specific code

### PR DESCRIPTION
**Pull request by ujvl**
Get rid of backwards compatible code

### CODE REVIEW

**Review by hhbyyh — APPROVED** [review]
LGTM. And shall we also update try_to_create_directory(directory_path, warn_if_exist=True) at https://github.com/ray-project/ray/blob/master/python/ray/utils.py#L548? even it's not in Tune.

File: python/ray/utils.py [hunk 1]
```diff
@@ -545,26 +545,14 @@ def try_make_directory_shared(directory_path):
...
-        except OSError as e:
-            if e.errno != errno.EEXIST:
-                raise e
-            if warn_if_exist:
-                logger = logging.getLogger("ray")
-                logger.warning(
-                    "Attempted to create '{}', but the directory already "
-                    "exists.".format(directory_path))
-
+    os.makedirs(directory_path, exist_ok=True)
```
Comment (richardliaw) [inline python/ray/utils.py]: cc @edoakes can you take a look at this change?

**Review by richardliaw — APPROVED** [review]
Looks good; will merge after eoakes takes a look.

**Review by edoakes — APPROVED** [review]
LGTM

**Comment by edoakes** [comment]
@ujvl any chance you want to do the same for the core ray python code? :)

**Comment by richardliaw** [comment]
Hm, maybe we can do that in a separate PR.

**Comment by richardliaw** [comment]
jenkins test tune

**Pull request merged**